src/Jsontracer/JsonTracerViewerPanel.ts

```diff
@@ -0,0 +1,126 @@
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * Copyright (c) Microsoft Corporation
+ *
+ * All rights reserved.
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+ * associated documentation files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+/*
+Some part of this code refers to
+https://github.com/microsoft/vscode-extension-samples/blob/2556c82cb333cf65d372bd01ac30c35ea1898a0e/custom-editor-sample/src/catScratchEditor.ts
+*/
+
+import * as vscode from 'vscode';
+import {getNonce} from '../Utils/external/Nonce';
+import {getUri} from '../Utils/external/Uri';
+
+
+/* istanbul ignore next */
+export class JsonTracerViewerPanel implements vscode.CustomTextEditorProvider {
+  private _disposables: vscode.Disposable[] = [];
+
+  public static readonly viewType = 'one.editor.jsonTracer';
+
+  public static register(context: vscode.ExtensionContext): void {
+    const provider = new JsonTracerViewerPanel(context);
+    const registrations = [
+      vscode.window.registerCustomEditorProvider(JsonTracerViewerPanel.viewType, provider, {
+        webviewOptions: {
+          retainContextWhenHidden: true,
+        },
+      })
+      // Add command registration here
+    ];
+
+    registrations.forEach(disposable => context.subscriptions.push(disposable));
+  };
+
+  constructor(private readonly context: vscode.ExtensionContext) {}
+
+  public async resolveCustomTextEditor(
+      document: vscode.TextDocument, webviewPanel: vscode.WebviewPanel,
+      _token: vscode.CancellationToken): Promise<void> {
+    await this.initWebview(document, webviewPanel.webview);
+    this.initWebviewPanel(document, webviewPanel);
```

Comment:
Q. 'await' not needed? It seems the function sets some contexts.

Reply:
It's because of `initWebview` :)
In the `initWebview` function, following `await` is used.

```ts
let html = Buffer.from(await vscode.workspace.fs.readFile(htmlUri)).toString();
```

Reply:
I don't mean initWebview functions, but initWebviewPanel...
I just found out that it's syncronous, so never mind 😄

Reply:
It doesn't look critical, and I believe that you already examined that, so never mind :-D